tests/data/test_deita.py

```diff
@@ -0,0 +1,93 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+from unittest import mock
+
+
+def test_format_dataset():
+    from lit_gpt.data.deita import format_dataset
+
+    data = [
+        {
+            "prompt": "prompt1",
+            "prompt_id": "1",
+            "messages": [
+                {"content": "question1", "role": "user"},
+                {"content": "response1", "role": "assistant"},
+                {"content": "question2", "role": "user"},
+                {"content": "response2", "role": "assistant"},
+            ],
+        },
+        {
+            "prompt": "prompt2",
+            "prompt_id": "2",
+            "messages": [
+                {"content": "question3", "role": "user"},
+                {"content": "response3", "role": "assistant"},
+                {"content": "question4", "role": "user"},
+                {"content": "response4", "role": "assistant"},
+            ],
+        },
+    ]
+
+    assert format_dataset(data, include_multi_turn_conversations=False) == [
+        {"instruction": "question1", "output": "response1", "input": ""},
+        {"instruction": "question3", "output": "response3", "input": ""},
+    ]
+    assert format_dataset(data, include_multi_turn_conversations=True) == [
+        {"instruction": "question1", "output": "response1", "input": ""},
+        {"instruction": "question2", "output": "response2", "input": ""},
+        {"instruction": "question3", "output": "response3", "input": ""},
+        {"instruction": "question4", "output": "response4", "input": ""},
+    ]
+
+
+@mock.patch("lit_gpt.data.deita.format_dataset")
+@mock.patch("datasets.load_dataset")
+def test_deita(_, format_dataset_mock, mock_tockenizer, tmp_path):
+    from lit_gpt.data import Deita, SFTDataset
+
+    format_dataset_mock.return_value = [
+        {"instruction": "inst1", "output": "out1"},
+        {"instruction": "inst2", "output": "out2"},
+        {"instruction": "inst3", "output": "out3"},
+    ]
+
+    deita = Deita(
+        num_workers=0,
+        download_dir=tmp_path,
+    )
+    deita.connect(mock_tockenizer, batch_size=2, max_seq_length=10)
+    deita.prepare_data()
+    deita.setup()
+
+    train_dataloader = deita.train_dataloader()
+    assert isinstance(train_dataloader.dataset, SFTDataset)
+    assert len(train_dataloader) == 2
+
+    val_dataloader = deita.val_dataloader()
+    assert isinstance(val_dataloader.dataset, SFTDataset)
+    assert len(val_dataloader) == 2
+
+
+# def test_deita(mock_tockenizer, deita_path):
```

Comment:
Do you intend to uncomment this before merging?

Reply:
No, because I couldn't make this test work. See https://github.com/Lightning-AI/lit-gpt/pull/986#issuecomment-1977790038

If we're happy with the current testing, we can remove the commented code. If there is a strong need to have an end2end test, I don't know how to do it.

Reply:
Let's remove it and merge this